pontoon/sync/vcs/repositories.py

```diff
@@ -40,8 +42,16 @@ def pull(self, source=None, target=None):
         code, output, error = execute(command, target)
 
         if code == 0:
+
             log.debug("Git: Repository at " + source + " updated.")
 
+            if branch:
+                command = ["git", "checkout", branch]
+                code, output, error = execute(command)
+
+                if code == 0:
+                    log.debug("Git: Branch " + branch + " checked out.")
+
         else:
             log.info("Git: " + unicode(error))
             log.debug("Git: Clone instead.")
```

Comment:
If the repo doesn't exist yet, we need to clone it. And again, switch to the right branch. Otherwise, on first sync Pontoon will synchronize with the default branch.